### PR TITLE
[feat][workflow] Upgrade to new version of documentation bot

### PR DESCRIPTION
**Pull request by maxsxu**
Master Issue: #15797

### Motivation

We have implemented the new documentation bot at https://github.com/apache/pulsar-test-infra/pull/43.
Now we can apply it to our workflow.

### Modifications

Replace the old action with the new version of documentation bot.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by Anonymitaet — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Review by urfreespace — APPROVED** [review]

**Comment by lhotari** [comment]
This seems to break the jobs. `Invalid repository 'apache/pulsar-test-infra/docbot'. Expected format {owner}/{repo}.` is the error message. @maxsxu did you test that this change works?

File: .github/workflows/ci-documentbot.yml [hunk 1]
```diff
@@ -17,33 +17,37 @@
...
     permissions:
       pull-requests: write 
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout action
+        uses: actions/checkout@v3
+        with:
+          repository: apache/pulsar-test-infra/docbot
+          ref: master
```
Comment (lhotari) [inline .github/workflows/ci-documentbot.yml:41]: this doesn't make sense.

**Comment by lhotari** [comment]
@Anonymitaet How was this tested? All jobs fail after this was merged.

File: .github/workflows/ci-documentbot.yml [hunk 2]
```diff
@@ -17,33 +17,37 @@
...
 
-      - uses: apache/pulsar-test-infra/doc-label-check@master
+      - name: Set up Go
+        uses: actions/setup-go@v3
         with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          label-pattern: '- \[(.*?)\] ?`(.+?)`' # matches '- [x] `label`'
+          go-version: 1.18
```
[the remark this answers is not in the document]
Comment (lhotari) [reply .github/workflows/ci-documentbot.yml:46]: got it, the action is a composite job so it needs it.

**Pull request merged**